keras/preprocessing/image.py

```diff
@@ -872,10 +872,10 @@ def __init__(self, x, y, image_data_generator,
                  data_format=None,
                  save_to_dir=None, save_prefix='', save_format='png'):
         if y is not None and len(x) != len(y):
-            raise ValueError('x (images tensor) and y (labels) '
+            raise ValueError('`x` (images tensor) and `y` (labels) '
                              'should have the same length. '
-                             'Found: x.shape = %s, y.shape = %s' %
-                             (np.asarray(x).shape, np.asarray(y).shape))
+                             'Found: x.shape = %d, y.shape = %d' %
+                             (len(x), len(y)))
 
```

Comment:
Note that `np.asarray` creates a new array. It is unnecessary to consume memory and time during just input spec checks.

Reply:
We want to print the full shapes for clarity.

Creating a new array in the exception is not an issue: we're interrupting execution anyway, so we don't care about resources consumption.

Reply:
Reverted, @fchollet.